engine/api/environment/dao_key.go

```diff
@@ -22,6 +23,9 @@ func InsertKey(db gorp.SqlExecutor, key *sdk.EnvironmentKey) error {
 	dbEnvironmentKey.Private = string(s)
 
 	if err := db.Insert(&dbEnvironmentKey); err != nil {
+		if errPG, ok := err.(*pq.Error); ok && errPG.Code == "23505" {
```

Comment:
string `23505` has 4 occurrences, make it a constant

Reply:
nice catch @golangci !